cmd/readiness/main.go

```diff
@@ -45,59 +44,38 @@ func init() {
 // Additionally if the previous check hasn't returned 'true' the "deadlock" case is checked to make sure the Agent is
 // not waiting for the other members.
 func isPodReady(conf config.Config) bool {
-	fd, err := os.Open(conf.HealthStatusFilePath)
+	healthStatus, err := conf.HealthStatusProvider.HealthStatus()
 	if err != nil {
-		logger.Warn("No health status file exists, assuming the Automation agent is old")
-		return true
-	}
-	defer fd.Close()
-
-	health, err := readAgentHealthStatus(fd)
-	if err != nil {
-		logger.Errorf("Failed to read agent health status file: %s", err)
-		// panicking allows to see the problem in the events for the pod (kubectl describe pod ..)
-		panic("Failed to read agent health status file: %s")
+		panic(err)
 	}
 
 	// The 'statuses' file can be empty only for OM Agents
-	if len(health.Healthiness) == 0 && !isHeadlessMode() {
+	if len(healthStatus.Healthiness) == 0 && !isHeadlessMode() {
 		logger.Info("'statuses' is empty. We assume there is no automation config for the agent yet.")
 		return true
 	}
 
-	// If the agent has reached the goal state - returning true
-	inGoalState, err := isInGoalState(health, conf)
+	// If the agent has reached the goal state
+	inGoalState, err := isInGoalState(healthStatus, conf)
 	if err != nil {
 		logger.Errorf("There was problem checking the health status: %s", err)
 		panic(err)
```

Comment:
if we change the above panic, maybe we can do the same thing here to maintain consistency. What do you think?